app/views/handlers/list_action.py

```diff
@@ -29,17 +29,14 @@ def is_location_valid(self):
         return True
 
     def get_previous_location_url(self):
-        if self._return_to:
-            block_id = self._return_to
-
-            if self._schema.is_block_valid(block_id):
-                section_id = self._schema.get_section_id_for_block_id(block_id)
-                return Location(section_id=section_id, block_id=block_id).url()
-
-        return self.parent_location.url()
+        block_id = self._request_args.get('previous') or self._request_args.get(
+            'return_to'
+        )
+        return self._get_location_url(block_id)
 
     def get_next_location_url(self):
-        return self.parent_location.url()
+        block_id = self._request_args.get('return_to')
```

Comment:
I can't follow what's going on here. Both the next and previous seem to be return_to if given?

Reply:
Isn't there a case where return_to may not be supplied?

Reply:
> I can't follow what's going on here. Both the next and previous seem to be return_to if given?

I guess this is actually what we want though 👍

Reply:
Yes. When coming from a summary both the previous link and submitting the change should return the user to the list summary.